changelogs/unreleased/3829-unclear-error-when-compiler-venv-has-incompatible-python-version.yml

```diff
@@ -0,0 +1,6 @@
+description: An exception is now raised when there is a mismatch between the python versions of the compiler venv and the active process
+change-type: patch
+destination-branches: [master]
```

Comment:
There is something I don't understand here. Currently only ISO4 uses a compiler venv. ISO5+ execute compiles using the process venv. As such I would assume we only need a patch for ISO4. But the ticket talks about `inmanta-core>=6` (which is ISO5) and this PR provides a solution for ISO6. Am I missing something here?

Reply:
`inmanta-core>=6` still provides the option to pass another compiler venv (used by pytest-inmanta among others). This ticket is about that. I admit it could have been more clear.

Reply:
Conclusion: This change has to go into all lines of development (master, iso5 and iso4).